crates/move-to-polka/src/linker.rs

```diff
@@ -119,6 +152,56 @@ pub fn create_blob(
     Ok(blob)
 }
 
+fn fetch_git_dep(
+    mapping: &mut Vec<String>,
+    dep_sources: &mut Vec<String>,
+    dep: &move_package::source_package::parsed_manifest::Dependency,
+    git_url: &str,
+) {
+    let path = Path::new("/tmp/move-deps");
+    create_dir_all(path).expect("Failed to create temporary directory for dependencies");
+    match gix::open(path) {
+        Ok(repo) => {
+            let remote = repo
+                .find_default_remote(Direction::Fetch)
+                .expect("Failed to find default remote")
+                .expect("No default remote found");
+
+            remote
+                .connect(Direction::Fetch)
+                .expect("Failed to connect to remote")
+                .prepare_fetch(Discard, gix::remote::ref_map::Options::default())
+                .expect("Failed to prepare fetch")
+                .with_shallow(Shallow::DepthAtRemote(NonZero::new(1).unwrap()))
+                .receive(Discard, &AtomicBool::new(false))
+                .expect("Failed to fetch from remote");
+        }
+        Err(_) => {
+            let mut prep = gix::prepare_clone(git_url, path)
+                .expect("Failed to prepare clone")
+                .with_shallow(Shallow::DepthAtRemote(NonZero::new(1).unwrap()));
+
+            let (mut checkout, _) = prep
+                .fetch_then_checkout(Discard, &AtomicBool::new(false))
+                .expect("Failed to fetch and checkout");
+            let (_, _) = checkout
+                .main_worktree(Discard, &AtomicBool::new(false))
+                .expect("Failed to checkout main worktree");
+        }
+    };
+    let git_info = dep.git_info.as_ref().unwrap();
+    let source = format!("/tmp/move-deps/{}/sources", git_info.subdir.display());
+    dep_sources.push(source);
+    if let Some(dep_mapping) = dep.subst.as_ref() {
+        for m in dep_mapping {
+            if let SubstOrRename::Assign(ref addr) = m.1 {
+                let mapping_str = format!("{}={}", m.0, addr.to_standard_string());
+                mapping.push(mapping_str);
+            }
+        }
```

Comment:
Would something like that work for better readability (m.0/m.1 say not much)? (haven't checked it in compiler 😄 )

```suggestion
        for (name, subst) in dep_mapping {
            if let SubstOrRename::Assign(ref addr) = subst {
                mapping.push(format!("{}={}", name, addr.to_standard_string()));
            }
        }
```